python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_asset_resources.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+
+from dagster import (
+    AssetGroup,
+    ResourceDefinition,
+    asset,
+    fs_io_manager,
+    io_manager,
+    resource,
+    with_resources,
+)
+
+#####################################
+# Resource definitions on assets
+#####################################
+
+
+def test_with_resources():
+    @resource
+    def my_resource():
+        ...
+
+    @asset(resource_defs={"foo": my_resource})
+    def asset1():
+        ...
+
+    @asset(required_resource_keys={"foo"})
+    def asset2():
+        ...
+
+    assert asset1 == with_resources([asset2], resource_defs={"foo": my_resource})[0]
```

Comment:
why are these equal? asset1 and asset2 have (probably) different implementations

Reply:
ah yeah - they're meant to have the same implementation. I was basically just trying to get across that there's no difference between an asset whose resource def is supplied when the asset is defined and an asset whose resource def is supplied using `with_resources`. I'll update this example to make it make more sense.